client/opentracing/src/test/java/org/hawkular/apm/client/opentracing/APMTracerReferenceTest.java

```diff
@@ -0,0 +1,423 @@
+
+        assertTrue(parentComponent.getCorrelationIds().isEmpty());
+        assertEquals(1, parentComponent.getNodes().size());
+        assertEquals(Component.class, parentComponent.getNodes().get(0).getClass());
+
+        Component childComponent = (Component) parentComponent.getNodes().get(0);
+
+        assertTrue(childComponent.getCorrelationIds().isEmpty());
+        assertEquals(0, childComponent.getNodes().size());
+    }
+
+    @Test
+    public void testSingleFollowsFromRef() {
+        TestTraceReporter reporter = new TestTraceReporter();
+        APMTracer tracer = new APMTracer(reporter);
+
+        Span parentSpan = tracer.buildSpan("ParentSpan")
+                .start();
+
+        parentSpan.finish();
+
+        Span followsFromSpan = tracer.buildSpan("FollowsFromSpan")
+                .addReference(References.FOLLOWS_FROM, parentSpan.context())
+                .start();
+        followsFromSpan.finish();
+
+        assertEquals(2, reporter.getTraces().size());
+
+        Trace parentTrace = reporter.getTraces().get(0);
+
+        assertEquals(1, parentTrace.getNodes().size());
+        assertEquals(Component.class, parentTrace.getNodes().get(0).getClass());
+
+        Component parentComponent = (Component) parentTrace.getNodes().get(0);
+
+        assertTrue(parentComponent.getCorrelationIds().isEmpty());
+        assertEquals(0, parentComponent.getNodes().size());
+
+        Trace followsFromTrace = reporter.getTraces().get(1);
+
+        assertEquals(1, followsFromTrace.getNodes().size());
+
+        // 'Consumer' introduced to link the followsFrom component to the referenced Span/Node
+        assertEquals(Consumer.class, followsFromTrace.getNodes().get(0).getClass());
+
+        Consumer followsFromConsumer = (Consumer) followsFromTrace.getNodes().get(0);
+
+        assertEquals(followsFromConsumer.getCorrelationIds().get(0),
+                new CorrelationIdentifier(Scope.CausedBy, parentTrace.getFragmentId() + ":0"));
+        assertEquals(1, followsFromConsumer.getNodes().size());
+        assertEquals(Component.class, followsFromConsumer.getNodes().get(0).getClass());
+
+        Component followsFromComponent = (Component) followsFromConsumer.getNodes().get(0);
+
+        assertTrue(followsFromComponent.getCorrelationIds().isEmpty());
+        assertEquals(0, followsFromComponent.getNodes().size());
+    }
+
+    /**
+     * This test shows multiple FollowsFrom references with just a single ChildOf(SpanContext).
+     */
+    @Test
+    public void testSingleChildOfSpanContextWithOtherFollowsFromRefs() {
+        TestTraceReporter reporter = new TestTraceReporter();
+        APMTracer tracer = new APMTracer(reporter);
+
+        Span refdSpan1 = tracer.buildSpan("ref1")
+                .start();
+        refdSpan1.finish();
+
+        Span refdSpan2 = tracer.buildSpan("ref2")
+                .start();
+        refdSpan2.finish();
+
+        Map<String, String> headers = new HashMap<>();
```

Comment:
Could you please define some method in this class that would return extracted context based on these constants? It's repeated multiple times.